src/main/java/edu/cornell/kfs/coa/dataaccess/impl/CuAccountDelegateGlobalDaoOjb.java

```diff
@@ -0,0 +1,20 @@
+package edu.cornell.kfs.coa.dataaccess.impl;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.kuali.kfs.coa.dataaccess.impl.AccountDelegateGlobalDaoOjb;
+
+import edu.cornell.kfs.coa.dataaccess.CuAccountDelegateGlobalDao;
+
+public class CuAccountDelegateGlobalDaoOjb extends AccountDelegateGlobalDaoOjb implements CuAccountDelegateGlobalDao {
+
+    @Override
+    public List<String> createSearchPatternsFromLockingRepresentations(List<String> lockingRepresentations) {
+        return lockingRepresentations.stream()
+                .map(this::convertForLikeCriteria)
+                .distinct()
```

Comment:
The duplication is just a result of how KualiCo creates the wildcarded versions of the locking representations; the non-wildcarded ones should already be unique. As per `AccountDelegateGlobalDaoOjb.convertForLikeCriteria()`, if the lock rep contains a doc type field, the whole part of the string after the field name will be replaced by a wildcard. This can result in a significant number of duplicate patterns, based on the lock-generating code in `AccountDelegateGlobalMaintainableImpl`.

Removing the duplicate patterns should greatly improve the efficiency of the constructed SQL query and also simplify its ORed section of the criteria.

Reply:
Thanks, that makes sense